src/configure/extract_from_c.ml

```diff
@@ -0,0 +1,118 @@
+(*
+ * Copyright (c) 2016 whitequark.
+ *
+ * This file is distributed under the terms of the MIT License.
+ * See the file LICENSE for details.
+ *)
+
+let getenv ~default name =
+  try Sys.getenv name
+  with Not_found -> default
+
+let nsplit sep str =
+  if str = "" then []
+  else
+    let rec loop acc pos =
+      if pos > String.length str then
+        List.rev acc
+      else
+        let i = try String.index_from str pos sep with Not_found -> String.length str in
+        loop (String.sub str pos (i - pos) :: acc) (i + 1)
+    in
+    loop [] 0
+
+let read_output program =
+  let input_filename = Filename.temp_file "ctypes_libffi_config" ".c" in
+  let channel = open_out input_filename in
+  output_string channel program;
+  close_out channel;
+  let output_filename = (Filename.chop_suffix input_filename ".c") ^ ".o" in
+  let cwd = Sys.getcwd () in
+  let cmd =
+    Printf.sprintf "%s ocamlc -verbose %s %s -c 1>&2"
+      (getenv ~default:"ocamlfind" "OCAMLFIND")
+      ((getenv ~default:"" "CFLAGS") |>
+       (nsplit ' ') |>
+       (List.map (fun s -> "-ccopt '"^s^"'")) |>
+       (String.concat " "))
+      input_filename
+  in
+  prerr_endline cmd;
+  Sys.chdir (Filename.dirname input_filename);
+  ignore (Sys.command cmd);
+  Sys.chdir cwd;
+  Sys.remove input_filename;
+  if not (Sys.file_exists output_filename) then
+    raise Not_found;
+  let channel = open_in_bin output_filename in
+  let length = in_channel_length channel in
+  let result = Bytes.create length in
+  really_input channel result 0 length;
+  close_in channel;
+  Sys.remove output_filename;
+  result
+
+let find_from haystack pos needle =
```

Comment:
There's a bug here: `pos` isn't used.  But it'd be simpler to implement `find_from` like this:
```ocaml
Str.(search_forward (regexp_string needle) haystack pos)
```